.grit/patterns/python/replace-tempfile-mktemp-with-tempfile-NamedTemporaryFile.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Replace `tempfile.mktemp` ⇒ `tempfile.NamedTemporaryFile`
+---
+
+Prefer using tempfile.NamedTemporaryFile instead. According to the official Python documentation, the tempfile.mktemp function is considered unsafe and should be avoided. This is because the generated file name may initially point to a non-existent file, and by the time you attempt to create it, another process may have already created a file with the same name, leading to potential conflicts.
+
+- [reference](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp)
+
+tags: #fix #good-practice
+
+```grit
+engine marzano(0.1)
+language python
+
+and {
+    `$tempfile.mktemp($params)` => `$tempfile.NamedTemporaryFile($params delete=False)`,
+    `$tempfile.mktemp($params)` where {
+        $params => `$params ,`,
+    }
+}
```

Comment:
Don't use `and` to combine side effects. It's meant to look for different patterns.

This is more idiomatic: https://app.grit.io/studio?key=m-vMP76mMeI9diT49f0nJ

Reply:
Yeah, I was trying something similar as my first solution. anyway updated it now